specs/MultiProfile.md

```diff
@@ -0,0 +1,255 @@
+    {
+        FeatureNotAvailable();
+        return S_OK;
+    }
+
+    Microsoft::WRL::ComPtr<ICoreWebView2ControllerOptions> options;
+    HRESULT hr = webViewEnvironment4->CreateCoreWebView2ControllerOptions(
+        m_webviewOption.profile.c_str(), m_webviewOption.isInPrivate, options.GetAddressOf());
+    if (hr == E_INVALIDARG)
+    {
+        ShowFailure(hr, L"Unable to create WebView2 due to an invalid profile name.");
+        CloseAppWindow();
+        return S_OK;
+    }
+    CHECK_FAILURE(hr);
+
+#ifdef USE_WEBVIEW2_WIN10
+    if (m_dcompDevice || m_wincompCompositor)
+#else
+    if (m_dcompDevice)
+#endif
+    {
+        CHECK_FAILURE(webViewEnvironment4->CreateCoreWebView2CompositionControllerWithOptions(
+            m_mainWindow, options.Get(),
+            Callback<ICoreWebView2CreateCoreWebView2CompositionControllerCompletedHandler>(
+                [this](
+                    HRESULT result,
+                    ICoreWebView2CompositionController* compositionController) -> HRESULT {
+                    auto controller =
+                        wil::com_ptr<ICoreWebView2CompositionController>(compositionController)
+                            .query<ICoreWebView2Controller>();
+                    return OnCreateCoreWebView2ControllerCompleted(result, controller.get());
+                })
+                .Get()));
+    }
+    else
+    {
+        CHECK_FAILURE(webViewEnvironment4->CreateCoreWebView2ControllerWithOptions(
+            m_mainWindow, options.Get(),
+            Callback<ICoreWebView2CreateCoreWebView2ControllerCompletedHandler>(
+                this, &AppWindow::OnCreateCoreWebView2ControllerCompleted)
+                .Get()));
+    }
+
+    return S_OK;
+}
+```
+
+## Access the profile property of WebView2
+
+### Win32 C++
+
+```cpp
+HRESULT AppWindow::OnCreateCoreWebView2ControllerCompleted(HRESULT result, ICoreWebView2Controller* controller)
+{
+    // ...
+
+    m_controller = controller;
+    
+    // Gets the webview object from controller.
+    wil::com_ptr<ICoreWebView2> coreWebView2;
+    CHECK_FAILURE(m_controller->get_CoreWebView2(&coreWebView2));
+
+    auto webview7 = coreWebView2.try_query<ICoreWebView2_7>();
+    if (webview7)
+    {
+        // Gets the profile property of webview.
+        wil::com_ptr<ICoreWebView2Profile> profile;
+        CHECK_FAILURE(webview7->get_Profile(&profile));
+
+        // Accesses the profile object.
+        BOOL inPrivateModeEnabled = FALSE;
+        CHECK_FAILURE(profile->get_IsInPrivateModeEnabled(&inPrivateModeEnabled));
+        wil::unique_cotaskmem_string profile_path;
+        CHECK_FAILURE(profile->get_ProfilePath(&profile_path));
+        std::wstring str(profile_path.get());
+        m_profileDirName = str.substr(str.find_last_of(L'\\') + 1);
+        
+        // update window title with m_profileDirName
+        UpdateAppTitle();
+
+        // update window icon
+        SetAppIcon(inPrivate);        
+    }
+  
+    // ...
+}
+```
```

Comment:
Apologies I didn't realize there was no C# sample code in the previous PR. Please provide corresponding C# sample code from our C# sample app.

Reply:
David, I have added the C# sample code. Please help me check it, thanks.

Reply:
👍